doc/source/ray-air/examples/torch_detection.md

```diff
@@ -0,0 +1,476 @@
+                "model": model.module.state_dict(),
+                "optimizer": optimizer.state_dict(),
+                "lr_scheduler": lr_scheduler.state_dict(),
+                "config": config,
+                "epoch": epoch,
+            }
+        )
+        session.report({}, checkpoint=checkpoint)
+```
+
+### Fine-tune the model
+
+
+Once you've defined the training loop, create a `TorchTrainer` and pass the training
+loop to the constructor. Then, call `TorchTrainer.fit` to train the model.
+
+```python
+from ray.air.config import ScalingConfig
+from ray.train.torch import TorchTrainer
+
+
+trainer = TorchTrainer(
+    train_loop_per_worker=train_loop_per_worker,
+    train_loop_config={
+        "batch_size": 2,
+        "lr": 0.02,
+        "epochs": 26,
+        "momentum": 0.9,
+        "weight_decay": 1e-4,
+        "lr_steps": [16, 22],
+        "lr_gamma": 0.1,
+    },
+    scaling_config=ScalingConfig(num_workers=8, use_gpu=True),
+    datasets={"train": train_dataset},
+    dataset_config={
+        # Don't augment test images. Only apply `per_epoch_preprocessor` to the train
+        # set.
+        "train": DatasetConfig(
+            per_epoch_preprocessor=per_epoch_preprocessor,
+        ),
+    }
+    preprocessor=preprocessor,
+)
+results = trainer.fit()
+```
+
+## Evaluate the model on test data
+
+### Generate predictions on the test data
+
+
+`Predictors` let you perform scalable [batch prediction](batch-prediction) and
+[online inference](air-serving-guide). To evaluate the model, you'll use
+`BatchPredictor` to perform inference in a distributed fashion.
+
+Create a `BatchPredictor` and pass `TorchDetectionPredictor` to the constructor. Then,
+call `BatchPredictor.predict` to detect objects in the test data.
+
+```python
+from ray.train.batch_predictor import BatchPredictor
+from ray.train.torch import TorchDetectionPredictor
+
+
+model = models.detection.fasterrcnn_resnet50_fpn(num_classes=21)
+predictor = BatchPredictor.from_checkpoint(results.checkpoint, TorchDetectionPredictor, model=model)
+
+predictions = predictor.predict(
+    test_dataset,
+    feature_columns=["image"],
+    keep_columns=["boxes", "labels"],
+    batch_size=4,
+    num_gpus_per_worker=1,
+)
+predictions
+```
+
+### Evaluate the model
+
+
+Once you've created the `predictions` dataset, iterate over the rows of the dataset
+and compute the accuracy of the model.
+
+```python
+from torchmetrics.detection.mean_ap import MeanAveragePrecision
+
+
+metric = MeanAveragePrecision()
+for row in predictions.iter_rows():
+    preds = [
+        {
+            "boxes": torch.as_tensor(row["pred_boxes"]),
+            "scores": torch.as_tensor(row["pred_scores"]),
+            "labels": torch.as_tensor(row["pred_labels"]),
+        }
+    ]
+    target = [
+        {
+            "boxes": torch.as_tensor(row["boxes"]),
+            "labels": torch.as_tensor(row["labels"]),
+        }
+    ]
+    metric.update(preds, target)
+
+metric.compute()
+```
```

Comment:
anything to show at the end? "We can see that our model has X% accuracy"

Reply:
`metric.compute()` returns the model accuracies. You'd see it in the Jupyter notebook.